src/coreclr/vm/debugdebugger.cpp

```diff
@@ -460,14 +460,17 @@ FCIMPL4(void, DebugStackTrace::GetStackFramesInternal,
         int iNumValidFrames = 0;
         for (int i = 0; i < data.cElements; i++)
         {
-            // The managed stacktrace classes always returns typical method definition, so we don't need to bother providing exact instantiation.
-            // Generics::GetExactInstantiationsOfMethodAndItsClassFromCallInformation(data.pElements[i].pFunc, data.pElements[i].pExactGenericArgsToken, &pExactMethod, &thExactType);
             MethodDesc* pFunc = data.pElements[i].pFunc;
-
-            // Strip the instantiation to make sure that the reflection never gets a bad method desc back.
-            if (pFunc->HasMethodInstantiation())
-                pFunc = pFunc->StripMethodInstantiation();
-            _ASSERTE(pFunc->IsRuntimeMethodHandle());
+            PTR_VOID pExactGenericArgsToken = data.pElements[i].pExactGenericArgsToken;
+            if (pFunc->HasClassOrMethodInstantiation() && pFunc->IsSharedByGenericInstantiations())
+            {
+                if (pExactGenericArgsToken != NULL)
+                {
+                    TypeHandle th;
+                    // TODO: an AV can happen here, not sure how to fix.
+                    Generics::GetExactInstantiationsOfMethodAndItsClassFromCallInformation(pFunc, pExactGenericArgsToken, &th, &pFunc);
```

Comment:
How can even an AV happen here?

Reply:
It is likely the same root cause as https://github.com/dotnet/runtime/pull/96440#discussion_r1441365564